site/_shortcodes/Example.js

```diff
@@ -17,11 +17,186 @@
 const fs = require('fs/promises');
 const path = require('path');
 const Prism = require('prismjs');
-const md = require('markdown-it')();
+const espree = require('espree');
+const execa = require('execa');
 
 const examples = [];
 let counter = {};
 
+class Example {
+  constructor(nunjucksEngine, context, raw) {
+    this.renderer = nunjucksEngine;
+
+    this.context = context;
+    this.raw = raw;
+
+    this.title = context.title;
+    this.slug = context.page.fileSlug;
+    counter[this.slug] = counter[this.slug] + 1 || 0;
+    this.id = `${this.slug}-${counter[this.slug]}`;
+
+    this.parsed = {
+      webComponents: this._parseHtml(),
+      react: this._parseJavaScript(),
+    }
+
+    this.previews = this._renderPreviews();
+    this.widget = this._renderWidget();
+  }
+
+  _extract(regExp, string) {
+    const match = regExp.exec(string);
+    return match && match[1] ? match[1] : null;
+  }
+
+  _parseCode(language, string) {
+    const code = {
+      all: this._extract(new RegExp('```' + language + '\n(.*?)\n```', 'gms'), string),
+    };
+
+    return code;
+  }
+
+  _parseHtml() {
+    const html = this._parseCode('html', this.raw);
+    if (!html.all) {
+      return null;
+    }
+
+    html.head = this._extract(/<head>(.*?)<\/head>/gms, html.all);
+    html.body = this._extract(/<body>(.*?)<\/body>/gms, html.all);
+
+    // Make sure head and body both don't contain any blank lines
+    // as they would create empty paragraphs in markdown
+    html.head = html.head.replace(/^\s*\n/gm, '').replace(/\s*$/, '');
```

Comment:
extract into a method

Reply:
By skipping markdown-it for the widgets that's not even needed anymore, removed 🙂